### PR TITLE
Allow removal of malicious whitelist members

### PR DESCRIPTION
**Pull request by LefterisJP**
- Add a function that will allow the guardian (soon to be renamed to
  curator) to remove addresses from the whitelist.

- Assert that the recipient of a proposal is in the whitelist at
  `executeProposal()`

The thinking behind this patch is that a party of malicious token
holders can cooperate with a whitelist member and rob the DAO of its
money.

The token holders can still split away, but those who don't manage to
split in time will stay with a powerless guardian (curator) who will be
unable to stop the proposal from executing. So in this case the lazy
voters would lose all of their money.

By allowing the removal of malicious actors from the whitelist and by
double-checking that someone is still on the whitelist at the time of
`executeProposal()` the guardian (curator) can still save the lazy voters.

### CODE REVIEW

**Comment by LefterisJP** [comment]
Related to #78

**Comment by LefterisJP** [comment]
A note about the function name. @GriffGreen proposed during a mumble session that adding/removing people from the whitelist could be a single function with a boolean argument. But I think having 2 separate functions is more user friendly and intuitive. It's open for discussion though.

File: DAO.sol [hunk 1]
```diff
@@ -484,6 +490,8 @@ contract DAO is DAOInterface, Token, TokenSale {
         if (now < p.votingDeadline  // has the voting deadline arrived?
             // Have the votes been counted?
             || !p.open
+            // is the recipient still on the whitelist?
+            || isRecipientAllowed(p.recipient)
```
Comment (CJentzsch) [inline DAO.sol]: so in this case, the voters would be blocked forever and the proposal deposit could never be returned.

**Comment by CJentzsch** [comment]
replaced by https://github.com/slockit/DAO/pull/84/files

**Pull request closed without merging**